river/tree/streaming_gradient_tree.py

```diff
@@ -0,0 +1,460 @@
+import abc
+import collections
+import functools
+import math
+import numbers
+import typing
+
+from scipy.stats import f as FTest
```

Comment:
`f` internally [uses](https://github.com/scipy/scipy/blob/ab1c0907fe9255582397db04592d6066745018d3/scipy/stats/_continuous_distns.py#L1938) `f_gen`, why not use this name instead?

Reply:
I vote to use `f_test` for clarity. What do you think?

Reply:
Edit: Let's go with `f_dist` since I am actually using the generator to calculate the CDF. Thanks for the suggestion.